rust/src/lib/unit_tests/ethernet.rs

```diff
@@ -393,3 +393,41 @@ fn test_mac_identifer_check_iface_type_also() {
     );
     assert_eq!(des_iface.base_iface().profile_name.as_deref(), Some("wan0"))
 }
+
+#[test]
+fn test_ignore_speed_duplex_when_auto_negotiate() {
+    let cur_ifaces: Interfaces = serde_yaml::from_str(
+        r"---
+        - name: eth1
+          type: ethernet
+          state: up
+          ethernet:
+            auto-negotiation: true
+            speed: 900
+            duplex: half
+        ",
+    )
+    .unwrap();
+
+    let des_ifaces: Interfaces = serde_yaml::from_str(
+        r"---
+        - name: eth1
+          type: ethernet
+          state: up
+          ethernet:
+            auto-negotiation: true
+            speed: 1000
+            duplex: full
+        ",
+    )
+    .unwrap();
+
+    let merged_ifaces = MergedInterfaces::new(
+        des_ifaces,
+        cur_ifaces.clone(),
+        Default::default(),
+        false,
+    )
+    .unwrap();
+    merged_ifaces.verify(&cur_ifaces).unwrap();
+}
```

Comment:
Nit, for a comprehensive test, maybe we should also validate when auto-negotiation is false, speed/duplex differences should cause verification failure, what do you think?